File: BackEnd_Learn/Java/jwt/src/main/java/com/example/jwt/Security/Jwt/JwtTokenFilter.java
package com.example.jwt.Security.Jwt;

import com.example.jwt.Security.Jwt.Auth.AuthDetail;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.filter.GenericFilterBean;
import org.springframework.web.filter.OncePerRequestFilter;

import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

@RequiredArgsConstructor
public class JwtTokenFilter extends GenericFilterBean {

    private final JwtTokenProvider jwtTokenProvider;

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        String token = jwtTokenProvider.ResolveToken((HttpServletRequest)request);
        if(token != null && jwtTokenProvider.validateToken(token)){
            Authentication auth = jwtTokenProvider.authentication(token);
            SecurityContextHolder.getContext().setAuthentication(auth);
        }
        chain.doFilter(request, response);
    }
}

File: BackEnd_Learn/kotlin/entity/src/main/kotlin/com/example/entity/security/jwt/JwtTokenFilter.kt
package com.example.entity.security.jwt

import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.filter.GenericFilterBean
import javax.servlet.FilterChain
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse
import javax.servlet.http.HttpServletRequest

class JwtTokenFilter(val jwtTokenProvider: JwtTokenProvider): GenericFilterBean() {


    override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain?) {
        request ?: throw Exception()
        val token: String? = jwtTokenProvider.resolveToken(request as HttpServletRequest)
        token?.let {
            if(jwtTokenProvider.validateToken(token)) {
                val auth: Authentication = jwtTokenProvider.authentication(token)
                SecurityContextHolder.getContext().authentication = auth
            }
        }
        chain?.doFilter(request, response)
//        var token: String? = jwtTokenProvider.resolveToken(request as HttpServletRequest)
//        try{
//
//            if (token != null && jwtTokenProvider.validateToken(token)) run {
//                val auth: Authentication = jwtTokenProvider.authentication(token)
//                SecurityContextHolder.getContext().authentication = auth
//            }
//        }

    }
}